### ISSUE TITLE
Middle click event is reported as right click

### ISSUE THREAD

**Issue by ianhi** [matplotlib/ipympl#254]
## Describe the issue

If you middle click on an ipympl canvas the event will be reported with mousebutton.RIGHT instead of MIDDLE.
```python
import ipywidgets as widget
import matplotlib.pyplot as plt
%matplotlib widget

out = widgets.Output()
@out.capture()
def callback(event):
    print(event.button)
    
plt.ioff(); fig,ax = plt.subplots(); plt.ion()
fig.canvas.mpl_connect('button_press_event', callback)
fs = 15
ax.text(0.1,0.1,'left click here', fontsize=fs)
ax.text(0.5,0.5,'middle click here', fontsize=fs)
ax.text(0.7,0.8,'right click here', fontsize=fs)
widgets.HBox([out,fig.canvas])
```
In the notebook:
![jupyter](https://user-images.githubusercontent.com/10111092/89341236-fe55bd80-d66e-11ea-91bb-d90337c64312.gif)

vs from ipython:
![ipython](https://user-images.githubusercontent.com/10111092/89341289-13cae780-d66f-11ea-9451-6cb9290a20fe.gif)

## Versions

```
 3.8.5 (default, Aug  3 2020, 16:08:50) 
[GCC 7.3.0]
ipympl version: 0.5.7
jupyter core     : 4.6.3
jupyter-notebook : 6.1.1
qtconsole        : not installed
ipython          : 7.17.0
ipykernel        : 5.3.4
jupyter client   : 6.1.6
jupyter lab      : 2.2.2
nbconvert        : 5.6.1
ipywidgets       : 7.5.1
nbformat         : 5.0.7
traitlets        : 4.3.3
Known nbextensions:
  config dir: /home/ian/anaconda3/envs/micro/etc/jupyter/nbconfig
    notebook section
      jupyter-matplotlib/extension  enabled 
      - Validating: OK
      nbdime/index  enabled 
      - Validating: OK
      jupyter-js-widgets/extension  enabled 
      - Validating: OK
JupyterLab v2.2.2
Known labextensions:
   app dir: /home/ian/anaconda3/envs/micro/share/jupyter/lab
        @axlair/jupyterlab_vim v0.12.4  enabled  OK
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        @jupyterlab/git v0.20.0  enabled  OK
        @telamonian/theme-darcula v2.0.0  enabled  OK
        jupyter-matplotlib v0.7.3  enabled  OK
        jupyterlab_vim-system-clipboard-support v0.2.0  enabled  OK
        nbdime-jupyterlab v2.0.0  enabled  OK
```

**Comment by ianhi** [comment]
Ooh ok here is why:
https://github.com/matplotlib/matplotlib/blob/ede8e566e5c7e7a12fdd66231ba02f363aab8913/lib/matplotlib/backends/backend_webagg_core.py#L263-L269
```python
        if button == 2:
            button = 3
```

I will open an issue on the main matplotlib repo

**Comment by ianhi** [comment]
should be fixed by the linked matplotlib PR assuming it is merged. That be a backwards incompatible change and I'm not sure if that would require significant waiting for a new major version.

**Comment by ianhi** [comment]
This will be fixed with the release of matplotlib 3.4
